airflow/providers/amazon/aws/operators/rds.py

```diff
@@ -672,6 +682,132 @@ def execute(self, context: Context) -> str:
         return json.dumps(delete_db_instance, default=str)
 
 
+class RdsStartDbOperator(RdsBaseOperator):
+    """
+    Starts an RDS DB instance / cluster
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:RdsStartDbOperator`
+
+    :param db_identifier: The AWS identifier of the DB to start
+    :param db_type: Type of the DB - either "instance" or "cluster" (default: "instance")
+    :param aws_conn_id: The Airflow connection used for AWS credentials. (default: "aws_default")
+    :param wait_for_completion:  If True, waits for DB to start. (default: True)
+    """
+
+    template_fields = ("db_identifier", "db_type")
+
+    def __init__(
+        self,
+        *,
+        db_identifier: str,
+        db_type: str = "instance",
+        aws_conn_id: str = "aws_default",
+        wait_for_completion: bool = True,
+        **kwargs,
+    ):
+        super().__init__(aws_conn_id=aws_conn_id, **kwargs)
+        self.db_identifier = db_identifier
+        self.db_type = db_type
+        self.wait_for_completion = wait_for_completion
+
+    def execute(self, context: Context) -> str:
+        self.db_type = RdsDbType(self.db_type)
+        start_db_response = self._start_db()
+        if self.wait_for_completion:
+            self._wait_until_db_available()
+        return json.dumps(start_db_response, default=str)
+
+    def _start_db(self):
+        self.log.info(f"Starting DB {self.db_type} '{self.db_identifier}'")
+        if self.db_type == RdsDbType.INSTANCE:
+            response = self.hook.conn.start_db_instance(DBInstanceIdentifier=self.db_identifier)
+        else:
+            response = self.hook.conn.start_db_cluster(DBClusterIdentifier=self.db_identifier)
+        return response
+
+    def _wait_until_db_available(self):
```

Comment:
Maybe we could use here already written `_await_status` method.
It will reduce the code and unify all operators within this file.
WDYT?

Reply:
@kazanzhy 
Thanks for your review. I agree, that would be more consistent. What concerns me is that when using `_await_status` we need to specify all possible `wait_statuses`. I observed the following states when creating an RDS instance, but [depending on the settings, there may be more.](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/accessing-monitoring.html) (And they might change in the future?)
- Creating
- Backing-up
- Configuring-enhanced-monitoring

Rather than listing every possible state or risk "missing one," I think using the builtin "waiter" implementation is more appropriate here. We only need to check for the "available" state.

@o-nikolas
I noticed you work at Amazon, is there anything you'd like to add to this discussion?

Reply:
I agree, we should use waiters as much as possible. They come with nice features such as retry with exponential backoff, ...

Reply:
Our preference for AWS code is to use the boto Waiters wherever possible. I don't have the context to know whether the existing `_await_status` method could be updated to use waiters or not, but that would be my preference rather than the other direction. But I also think that's out of scope for this PR. So how you've got it now is fine, IMHO.

Reply:
I'm confused on why the _await_status is on operator level rather than on Hook level.

> What concerns me is that when using _await_status we need to specify all possible wait_statuses

Why? We need to wait for `Available` status only aren't we?

Reply:
> I'm confused on why the _await_status is on operator level rather than on Hook level.

Agreed, I think we should ultimately deprecate this thing in favour of boto waiters (and some shared hook method if there are some situations where waiters don't suffice). But I still think that work is out of scope for this PR.

Reply:
@o-nikolas can you please create a followup task and link to this discussion?
The `_await_status` is private function. we can remove it / change it as we see fit - it's not a breaking change.

Reply:
Yupp, will do :+1:

Reply:
#27096

Reply:
Just dogpiling on the "we should be using waiters everywhere" train.  There are a few other places (The EKS operators I wrote before I knew about the boto waiters, for example) in the package where we have custom wait methods but the official boto ones would be cleaner and easier alternatives.  It might make a nice task/project for new contributors to go through and see where they can be swapped out?

Reply:
@eladkal I remember that implemented `_await_status` to use the single method for different operators. I was the best solution in my opinion at that time. I knew about waiters but I guess there weren't all waiters that I needed.

I agree with @ferruzzi , let's use `_await_status` in this PR and then refactor the module and add waiters where it's possible. WDYT?

Reply:
Thank you all for the feedback. Consensus:
* Leave current changes as-is
* Refactor waiting logic in [#27096](https://github.com/apache/airflow/issues/27096)